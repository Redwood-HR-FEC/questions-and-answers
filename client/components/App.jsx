/* eslint-disable react/prefer-stateless-function */
import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import Votes from './Votes.jsx';

const axios = require('axios');


const GlobalStyle = createGlobalStyle`
@font-face {
  font-family: "Amazon Ember";
  src: url("https://m.media-amazon.com/images/G/01/AUIClients/AmazonUIFont-amazonember_rg-cc7ebaa05a2cd3b02c0929ac0475a44ab30b7efa._V2_.woff2");
}
@font-face {
  font-family: "Amazon Ember";
  font-weight: 700;
  src: url("https://m.media-amazon.com/images/G/01/AUIClients/AmazonUIFont-amazonember_bd-46b91bda68161c14e554a779643ef4957431987b._V2_.woff2");
}
`;
const Wrapper = styled.div`
  font-size: 13px;
  line-height: 19px;
  color: #111;
  font-family: "Amazon Ember",Arial,sans-serif;
  box-sizing:border-box;
  color:rgb(17, 17, 17);
  display:block;
  height:144px;
  line-height:19px;
  margin-bottom:14px;
  overflow-wrap:break-word;
  position:relative;
  text-size-adjust:100%;
  width:800px;
`;


class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      questions: [],
    };
    this.changeVotes = this.changeVotes.bind(this);
    this.fetchData = this.fetchData.bind(this);
  }

  componentDidMount() {
    const id = window.location.pathname.slice(1, -1);
    this.fetchData(id);
  }

  fetchData(id) {
    axios.get(`/questions/${id}`)
      .then((response) => {
        // console.log(response.data);
        this.setState({ questions: response.data });
      })
      .catch((error) => {
        console.log(error);
      });
  }

  changeVotes(vote, id, productId) {
    axios.patch('/questions', { vote, id })
      .then((response) => {
        this.fetchData(Number(productId));
      })
      .catch((error) => {
        console.log(error);
      });
  }

  render() {
    const { questions } = this.state;
    return (
      <Wrapper>
        <GlobalStyle />
        <Votes data={questions} changeVote={this.changeVotes} />
      </Wrapper>
    );
  }
}

export default App;
