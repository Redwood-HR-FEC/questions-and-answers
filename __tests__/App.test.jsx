import React from 'react';
import { shallow } from 'enzyme';
import axios from 'axios';
import App from '../client/components/App';


describe('Unit Tests for App', () => {
  jest.mock('axios', () => {
    const data = [{
      "_id" : "5eb0aa58fdb2ea2b7daec7f3",
      "answers" : [
        {
          "username" : "Ahmad.Bruen",
          "answer" : "Ergonomic green workforce Row 24/365 initiative Divide JBOD Money Market Account circuit Ergonomic Wooden Shoes Ergonomic backing up Handcrafted Radial wireless hub optical Up-sized Aruba",
          "date_posted" : "2019-12-07T17:14:54.404Z",
        },
        {
          "username" : "Vivien75",
          "answer" : "Automated deposit silver Strategist Division Berkshire Business-focused Montana deposit Bahamas Rustic Plastic relationships integrated Oklahoma collaborative Chips reboot primary Lebanese Pound",
          "date_posted" : "2019-10-24T15:58:43.321Z",
        }
      ],
      "product_id" : "068",
      "questions" : "tan Center Kansas Comoro Franc Developer sky blue Wooden open-source Designer Lead",
      "question_url" : "http://larissa.name",
      "votes" : 26,
      "__v" : 0,
    }];

    return {
      get: jest.fn(() => Promise.resolve(data)),
    };
  });

  it("renders correctly", () => {
    const wrapper = shallow(
      <App />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('fetches a list of data', () => {
    const getSpy = jest.spyOn(axios, 'get');
    const wrapper = shallow(
      <App />,
    );
    expect(getSpy).toBeCalled();
  });

  it('calls componentDidMount', () => {
    jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = shallow(<App />);
    expect(App.prototype.componentDidMount.mock.calls.length).toBe(1);
  });

  test('should render the App component on the screen', () => {
    const wrapper = shallow(<App />);
    expect(wrapper).toExist();
  });
});
