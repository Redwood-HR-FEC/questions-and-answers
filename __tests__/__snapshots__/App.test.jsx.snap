// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit Tests for App renders correctly 1`] = `
<App__Wrapper>
  <App__H2>
    Customer questions & answers
  </App__H2>
  <Search />
  <Memo(GlobalStyleComponent) />
  <Votes
    changeVote={[Function]}
    data={Array []}
  />
</App__Wrapper>
`;
